From 191aeecea3255c5a389f1be56637e4bf5b20f24c Mon Sep 17 00:00:00 2001
From: berbascum <berbascum@ticv.cat>
Date: Mon, 12 Aug 2024 01:36:05 +0200
Subject: [PATCH 40/60] Use lineageos 18.1 build tools and remove Droidian gcc
 as incompatible with some kernel configurations.

Also reverted the TRIPLET to
android- again.

Clang downgraded to 12.0 from Droidian
---
 debian/kernel-info.mk                       | 13 +++--
 droidian/scripts/compile-droidian-kernel.sh | 53 +++++++++++++++++++++
 2 files changed, 63 insertions(+), 3 deletions(-)

diff --git a/debian/kernel-info.mk b/debian/kernel-info.mk
index 8d5503288af6..0eb4876183ba 100644
--- a/debian/kernel-info.mk
+++ b/debian/kernel-info.mk
@@ -188,7 +188,7 @@ BUILD_CROSS = 1
 # (Cross-build only) The build triplet to use. You'll probably want to
 # use aarch64-linux-android- if building Android kernels.
 # might need to use aarch64-linux-gnu- with newer kernels such as GKI
-BUILD_TRIPLET = aarch64-linux-gnu-
+BUILD_TRIPLET = aarch64-linux-android-
 
 # (Cross-build only) The build triplet to use with clang. You'll probably
 # want to use aarch64-linux-gnu- here.
@@ -202,12 +202,19 @@ BUILD_LLVM = 1
 
 # Extra paths to prepend to the PATH variable. You'll probably want
 # to specify the clang path here (the default).
-BUILD_PATH = /usr/lib/llvm-android-14.0-r450784d/bin
+# Paths for lineage kernel build
+BUILD_PATH = /opt/build-tools/linux-x86/bin:/opt/android_prebuilts_gcc_linux-x86_aarch64_aarch64-linux-android-4.9/aarch64-linux-android/bin:/opt/android_prebuilts_gcc_linux-x86_aarch64_aarch64-linux-android-4.9/bin:/usr/lib/llvm-android-12.0-r416183b/bin:/opt/android_prebuilts_tools-lineage/linux-x86/bin
+
+# 10 r370808
+# 12 r416183b
+# 14 r450784d
 
 # Extra packages to add to the Build-Depends section. Mainline builds
 # can have this section empty, unless cross-building.
+
 # The default is enough to install the Android toolchain, including clang.
-DEB_TOOLCHAIN = vim, linux-packaging-snippets, linux-initramfs-halium-generic:arm64, binutils-aarch64-linux-gnu, clang-android-14.0-r450784d, gcc-4.9-aarch64-linux-android, g++-4.9-aarch64-linux-android, libgcc-4.9-dev-aarch64-linux-android-cross
+DEB_TOOLCHAIN = linux-initramfs-halium-generic:arm64, binutils-aarch64-linux-gnu, clang-android-12.0-r416183b
+#DEB_TOOLCHAIN = vim, linux-packaging-snippets, linux-initramfs-halium-generic:arm64, binutils-aarch64-linux-gnu, clang-android-10.0-r370808, gcc-4.9-aarch64-linux-android, g++-4.9-aarch64-linux-android, libgcc-4.9-dev-aarch64-linux-android-cross
 
 # Where we're building on
 DEB_BUILD_ON = amd64
diff --git a/droidian/scripts/compile-droidian-kernel.sh b/droidian/scripts/compile-droidian-kernel.sh
index 721c99ab19c0..c8fc332480c6 100755
--- a/droidian/scripts/compile-droidian-kernel.sh
+++ b/droidian/scripts/compile-droidian-kernel.sh
@@ -1,5 +1,58 @@
 #!/bin/bash
 
+START_DIR="/buildd/sources"
+ROOTDIR="/opt"
+
+fn_install_prereqs() {
+ apt-get install linux-packaging-snippets bc bison build-essential ccache curl flex git git-lfs gnupg gperf imagemagick libelf-dev  libncurses5-dev libssl-dev libxml2 libxml2-utils lzop pngcrush rsync schedtool squashfs-tools xsltproc zip zlib1g-dev python3 python-is-python3
+# libsdl1.2-dev
+}
+
+fn_enable_ccache() {
+    ## Use ccache tu speedup the build
+    export USE_CCACHE=1
+    export CCACHE_EXEC=/usr/bin/ccache
+    ccache -M 10G
+    ccache -o compression=false
+}
+
+fn_install_toolchains() {
+    cd /opt
+
+    ## If kernel is not previously downloaded.
+    # git clone https://github.com/LineageOS/android_kernel_device
+    #
+    # gcc
+    git clone -b lineage-18.1 https://github.com/LineageOS/android_prebuilts_gcc_linux-x86_aarch64_aarch64-linux-android-4.9.git
+    #
+    # clang
+#    git clone https://github.com/LineageOS/android_prebuilts_clang_kernel_linux-x86_clang-r416183b.git
+    #
+    # wireguard kernel module
+    # git clone https://github.com/WireGuard/wireguard-linux-compat.git
+    #
+    # built-tools
+
+# android-11.0.0_r0.100
+# android-13.0.0_r0.117
+    git clone -b android-11.0.0_r0.100 https://android.googlesource.com/kernel/prebuilts/build-tools
+    git clone -b lineage-18.1 https://github.com/LineageOS/android_prebuilts_tools-lineage.git
+}
+
+fn_invert_PATH_kernel_snippet() {
+    ## Patch releng kernel-snippet.mk
+    ## To use the lineage toolchain, the FULL_PATH var needs to be defined with the PATH var at the beguin
+    sed -i 's|FULL_PATH = $(BUILD_PATH):$(CURDIR)/debian/path-override:${PATH}|FULL_PATH = ${PATH}:$(BUILD_PATH):$(CURDIR)/debian/path-override|g' /usr/share/linux-packaging-snippets/kernel-snippet.mk
+}
+
+fn_install_prereqs
+fn_enable_ccache
+
+## CUSTOM TOOLCHAIN
+fn_invert_PATH_kernel_snippet
+fn_install_toolchains
+## Paths defined in kernel-info.mk
+
 chmod +x /buildd/sources/debian/rules
 cd /buildd/sources
 rm -f debian/control
-- 
2.39.2

